File: PYTHON/staff.py
# staff.py
# Staff mewarisi Person

from person import Person

class Staff(Person):
    def __init__(self, id="", nama="", usia=0, posisi="", gaji=0.0, shift=""):
        super().__init__(id, nama, usia)
        self.posisi = posisi
        self.gaji = gaji
        self.shift = shift

    def get_posisi(self): return self.posisi
    def set_posisi(self, posisi): self.posisi = posisi

    def get_gaji(self): return self.gaji
    def set_gaji(self, gaji): self.gaji = gaji

    def get_shift(self): return self.shift
    def set_shift(self, shift): self.shift = shift

File: JAVA/Staff.java
// Staff.java
// Staff mewarisi Person (hierarchical inheritance)

public class Staff extends Person {
    private String posisi;
    private double gaji;
    private String shift;

    public Staff() {
        super();
        this.posisi = "";
        this.gaji = 0.0;
        this.shift = "";
    }

    public Staff(String id, String nama, int usia, String posisi, double gaji, String shift) {
        super(id, nama, usia);
        this.posisi = posisi;
        this.gaji = gaji;
        this.shift = shift;
    }

    // Getter
    public String getPosisi() { return posisi; }
    public double getGaji() { return gaji; }
    public String getShift() { return shift; }

    // Setter
    public void setPosisi(String posisi) { this.posisi = posisi; }
    public void setGaji(double gaji) { this.gaji = gaji; }
    public void setShift(String shift) { this.shift = shift; }
}